File: Testing/Code for Testing/Basic/JS/T402_arrays/original.c
#include <stdio.h>

void print_array(int arr[], int count) {
    int i = 0;
    while (i < count) {
        if (i > 0) {
            printf(" ");
        }
        printf("%d", arr[i]);
        i = i + 1;
    }
    printf("\n");
}

int array_sum(int arr[], int count) {
    int total = 0;
    int i = 0;
    while (i < count) {
        total = total + arr[i];
        i = i + 1;
    }
    return total;
}

int array_max(int arr[], int count) {
    int result = arr[0];
    int i = 1;
    while (i < count) {
        if (arr[i] > result) {
            result = arr[i];
        }
        i = i + 1;
    }
    return result;
}

int array_min(int arr[], int count) {
    int result = arr[0];
    int i = 1;
    while (i < count) {
        if (arr[i] < result) {
            result = arr[i];
        }
        i = i + 1;
    }
    return result;
}

int count_above(int arr[], int count, int threshold) {
    int result = 0;
    int i = 0;
    while (i < count) {
        if (arr[i] > threshold) {
            result = result + 1;
        }
        i = i + 1;
    }
    return result;
}

void reverse_array(int arr[], int count) {
    int lo = 0;
    int hi = count - 1;
    while (lo < hi) {
        int temp = arr[lo];
        arr[lo] = arr[hi];
        arr[hi] = temp;
        lo = lo + 1;
        hi = hi - 1;
    }
}

void bubble_sort(int arr[], int count) {
    int i = 0;
    while (i < count - 1) {
        int j = 0;
        while (j < count - 1 - i) {
            if (arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
            j = j + 1;
        }
        i = i + 1;
    }
}

int main() {
    int data[] = {64, 34, 25, 12, 22, 11, 90, 45};
    int count = 8;

    printf("original: ");
    print_array(data, count);
    printf("sum: %d\n", array_sum(data, count));
    printf("max: %d\n", array_max(data, count));
    printf("min: %d\n", array_min(data, count));
    printf("above 30: %d\n", count_above(data, count, 30));

    reverse_array(data, count);
    printf("reversed: ");
    print_array(data, count);

    bubble_sort(data, count);
    printf("sorted: ");
    print_array(data, count);
    printf("sum after sort: %d\n", array_sum(data, count));

    return 0;
}

File: Testing/Code for Testing/Intermediate/Python/T302_stats/original.py
"""Basic statistics functions — mean, min, max, variance."""


def array_sum(data: list, count: int) -> int:
    total: int = 0
    i: int = 0
    while i < count:
        total = total + data[i]
        i = i + 1
    return total


def array_min(data: list, count: int) -> int:
    result: int = data[0]
    i: int = 1
    while i < count:
        if data[i] < result:
            result = data[i]
        i = i + 1
    return result


def array_max(data: list, count: int) -> int:
    result: int = data[0]
    i: int = 1
    while i < count:
        if data[i] > result:
            result = data[i]
        i = i + 1
    return result


def array_range(data: list, count: int) -> int:
    return array_max(data, count) - array_min(data, count)


def bubble_sort(data: list, count: int) -> None:
    i: int = 0
    while i < count - 1:
        j: int = 0
        while j < count - 1 - i:
            if data[j] > data[j + 1]:
                temp: int = data[j]
                data[j] = data[j + 1]
                data[j + 1] = temp
            j = j + 1
        i = i + 1


def median_val(data: list, count: int) -> int:
    bubble_sort(data, count)
    return data[count // 2]


def main() -> int:
    data: list = [23, 7, 42, 15, 8, 31, 19, 3, 50, 12]
    count: int = 10

    print(array_sum(data, count))
    print(array_min(data, count))
    print(array_max(data, count))
    print(array_range(data, count))

    # Median (sorts the array)
    print(median_val(data, count))

    # After sort, check first and last
    print(data[0])
    print(data[9])

    # Second dataset
    data2: list = [100, 200, 300, 400, 500]
    count2: int = 5
    print(array_sum(data2, count2))
    print(array_min(data2, count2))
    print(array_max(data2, count2))

    return 0


if __name__ == "__main__":
    main()
